### DISCUSSION TITLE
Application specific password usage in deliver

### DISCUSSION THREAD

**Discussion by mmiedlarz** [fastlane/fastlane#19113] [Q&A] [answered]
Hi there, 
In my project I'm currently using `upload_to_testflight` action with `apple_id` and `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD` arguments, and it works on CI without saving my account password anywhere on CI. 
Now I want to add `deliver` action to upload some screenshots and app description automatically (without .ipa) but I can't make it work. I'm not sure how I can pass `apple_id` param (is it called `app` in deliver?) and current documentation does not cover it clearly (just found [mention here](https://docs.fastlane.tools/best-practices/continuous-integration/#application-specific-passwords) that it should be possible).
My question is: Is it possible to use `deliver` without `FASTLANE_PASSWORD` env? Or is `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD` used only to bypass 2FA?

**Comment by lucgrabowski** [comment, marked as answer]
Hi @mmiedlarz,
`FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD` is used only to upload builds to AppStoreConnect, it cannot be used to upload other things, as written for method 3 in https://docs.fastlane.tools/getting-started/ios/authentication 
> The application specific password will not work if your action usage does anything else than uploading the binary, e.g. updating any metadata like setting release notes or distributing to testers, etc. For those actions, you must use one of the other methods.

If you want to pass Apple Id to `deliver`, you can do it with `username` parameter
https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/runner.rb#L28-L45 as it's used to login to AppStoreConnect.

To use `deliver` to upload metadata and screenshots you can check out other authentication methods in
https://docs.fastlane.tools/getting-started/ios/authentication/, e.g.
> Method 1: App Store Connect API key (recommended)

**Reply by mmiedlarz** [reply]
Hey @lucgrabowski,
Ahh I see now. Sorry I had to miss a comment about binary only. Will use different auth method then :)
Thanks for a quick response!

**Reply by lucgrabowski** [reply]
You're welcome ;)
